### PR TITLE
Progress on #5440.

### PR DESCRIPTION
**Pull request by bakercp**
Adds better backward compatibility during the `glm::vec*` / `ofVec*` transition.

### CODE REVIEW

File: libs/openFrameworks/3d/ofMesh.h [hunk 1]
```diff
@@ -654,6 +654,21 @@ class ofMesh_{
 
 	/// \}
 
+#ifndef OF_USE_LEGACY_MESH
+	OF_DEPRECATED_MSG("Use glm::vec2.", ofMesh_(ofPrimitiveMode mode, const vector<ofVec2f>& verts));
```
Comment (bakercp) [inline libs/openFrameworks/3d/ofMesh.h]: Well, technically, we aren't _adding_ new deprecated methods ... these are all methods that currently exist in 0.9.* and have not been gracefully deprecated, but rather hidden behind a somewhat unintuitive `OF_USE_LEGACY_MESH` def.  I say unintuitive based on my experiences with many students this past semester trying to mix old and new addons. Huge amounts of time were consumed trying to figure out the right combination of using  `OF_USE_LEGACY_MESH` or converting addons, to glm.  

So, basically I'd strongly argue for formally deprecating these by default to ease the transition, especially with the very heterogeneous state of addons.

**Comment by bakercp** [comment]
@arturoc Do you have additional thoughts on this?
